fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeMetadata.java

```diff
@@ -70,6 +105,83 @@ public List<String> listPartitionNames(String databaseName, String tableName, lo
         return deltaOps.getPartitionKeys(databaseName, tableName);
     }
 
+    @Override
+    public List<RemoteFileInfo> getRemoteFileInfos(Table table, List<PartitionKey> partitionKeys,
+                                                   long snapshotId, ScalarOperator operator,
+                                                   List<String> fieldNames, long limit) {
+        DeltaLakeTable deltaLakeTable = (DeltaLakeTable) table;
+        RemoteFileInfo remoteFileInfo = new RemoteFileInfo();
+        String dbName = deltaLakeTable.getDbName();
+        String tableName = deltaLakeTable.getTableName();
+        PredicateSearchKey key = PredicateSearchKey.of(dbName, tableName, snapshotId, operator);
+
+        triggerDeltaLakePlanFilesIfNeeded(key, table, operator, null);
+
+        List<Row> scanTasks = splitTasks.get(key);
+        if (scanTasks == null) {
+            throw new StarRocksConnectorException("Missing iceberg split task for table:[{}.{}]. predicate:[{}]",
+                    dbName, tableName, operator);
+        }
+
+        List<RemoteFileDesc> remoteFileDescs = Lists.newArrayList(
+                DeltaLakeRemoteFileDesc.createDeltaLakeRemoteFileDesc(scanTasks));
+        remoteFileInfo.setFiles(remoteFileDescs);
+        return Lists.newArrayList(remoteFileInfo);
+    }
+
+    private void triggerDeltaLakePlanFilesIfNeeded(PredicateSearchKey key, Table table, ScalarOperator operator,
+                                                   Tracers tracers) {
```

Comment:
tracers is used for `parallelPrepare`. you could delete it.

Reply:
Fixed